com.unity.netcode.gameobjects/Runtime/Core/NetworkObject.cs

```diff
@@ -712,17 +722,33 @@ private void OnTransformParentChanged()
                 m_LatestParent = null;
             }
 
+            // We always set this true to be able to apply the parenting
+            // If it is being de-parented, then m_IsReparented will be set
+            // to false.
+            // TODO: Determine if m_IsReparented is still needed.
```

Comment:
Will look further into this.

Reply:
So... this was my conundrum:
m_IsReparented was never getting set to false once it was set (server-side) and so once something was parented then it would always be true.  So the two "semi-hidden" places to look at would be `NetworkObject.GetMessageSceneObject` and `NetworkSpawnManager.OnDespawnObject` that both use the `NetworkObject.GetNetworkParenting`.

Both uses of "isReparented" depend upon the latestParent having a value.  Which really, we could remove m_IsReparented and just determine if it has a parent or not (without the reparented) as both cases are just trying to determine if it needs to set a parent or not.

Let me make some adjustments and get rid of the TODOs

Reply:
I went ahead and removed the m_IsReparented